src/v/cluster/archival_metadata_stm.cc

```diff
@@ -123,11 +125,14 @@ ss::future<std::error_code> archival_metadata_stm::do_add_segments(
     }
 
     auto batch = std::move(b).build();
-    auto result = co_await _raft->replicate(
+    auto fut = _raft->replicate(
       _insync_term,
       model::make_memory_record_batch_reader(std::move(batch)),
       raft::replicate_options{raft::consistency_level::quorum_ack});
 
+    auto deadline = ss::lowres_clock::now() + rc_node.get_timeout();
```

Comment:
nit: maybe we can simply expose the deadline value stored in the `rc_node` object and avoid time arithmetic.

Reply:
done